JavaFXUserInterface/src/main/java/controllers/AssetsController.java

```diff
@@ -190,99 +213,162 @@ private void sortingSetUp() {
         });
     }
 
+    /**
+     * Live search for assets in the assets list based on the input from the search bar.
+     * When a search has a match, all systems panes are cleared and generated again with the appropriate assets.
+     * When a search has no match, all system panes are cleared effectively showing no assets.
+     *
+     * @author Najim
+     */
+    private void searchAssets() {
+        searchedAssets = FXCollections.observableArrayList();
+        search.textProperty().addListener((observable, oldValue, newValue) -> {
+            searchedAssets.clear();
+            if (!newValue.equals("")) {
+                assets.stream().filter(o -> StringUtils.containsIgnoreCase(o.getRul().get(), newValue) || StringUtils.containsIgnoreCase(o.getName(), newValue)
```

Comment:
I was simply wondering if it wouldn't have been easier and more readable to just iterate through each asset's member variables and check the StringUtils.containsIgnoreCase(...) since the calling function is used and reused. It might be more complicated as well. I am just asking.

Reply:
Doing it this way or iterating through each asset's member variable would not change much except for readability and understanding of the code. Although, Iterating through the assets might result in more lines of code. 
I have added comments to explain what asset.stream().filter() does

Reply:
Yeah, I was talking about readability and understanding of the code. But I wasn't sure what was more readable between the stream and for/if-else statements. This way is fine.